File: tests/conformance/fixtures/python/01-basic.py
from sykli import Pipeline

p = Pipeline()

p.task("lint").run("npm run lint").inputs("**/*.ts")

p.task("test").run("npm test").after("lint") \
    .inputs("**/*.ts", "**/*.test.ts").timeout(120)

p.task("build").run("npm run build").after("test")

p.emit()

File: tests/conformance/fixtures/rust/01-basic.rs
use sykli::Pipeline;

fn main() {
    let mut p = Pipeline::new();
    p.task("lint").run("npm run lint").inputs(&["**/*.ts"]);
    p.task("test").run("npm test").after(&["lint"]).inputs(&["**/*.ts", "**/*.test.ts"]).timeout(120);
    p.task("build").run("npm run build").after(&["test"]);
    p.emit();
}
